samples/crypto/test_sample/README.rst

```diff
@@ -0,0 +1,258 @@
+   Make sure the test vector verdict is "passed"::
+
+      ***************************************
+      All Tests Done
+      X test vectors passed
+      0 test vectors failed
+      ***************************************
+#.
+
+Additional test cases and test vectors
+======================================
+
+Test cases and test vectors can be added to the test suite either by including additional source files or by extending the existing files.
+
+
+Test case
+---------
+
+A test case is a function designed to verify parts of the functionality of an nrf_crypto operation.
+Most nrf_crypto operations, like hash calculations and ECDH, have multiple test cases to be able to cover all features.
+A typical test case loops over all related test vectors, and logs the verdict of each.
+
+An example of a hash test case in a simplified form is shown below::
+
+   NRF_SECTION_DEF(test_vector_hash_data, test_vector_hash_t);
+   #define TEST_VECTOR_GET(i)      NRF_SECTION_ITEM_GET(test_vector_hash_data, test_vector_hash_t, (i))    /**< Get number of hash test vectors. */
+   #define TEST_VECTOR_COUNT       NRF_SECTION_ITEM_COUNT(test_vector_hash_data, test_vector_hash_t)       /**< Get test vector reference from array of test vectors. */
+
+   ret_code_t exec_test_case_hash(test_info_t * p_test_info)
+   {
+       uint32_t i;
+       ret_code_t err_code;
+       uint32_t hash_test_vector_count = TEST_VECTOR_COUNT;
+
+       for (i = 0; i < hash_test_vector_count; i++)
+       {
+           test_vector_hash_t * p_test_vector = TEST_VECTOR_GET(i);
+
+           // TODO: Add test code here.
+
+           // Will goto exit_test_vector if fails.
+           TEST_VECTOR_ASSERT(err_code);
+
+           p_test_info->tests_passed++;
+
+   exit_test_vector: ;
+       }
+       return NRF_SUCCESS;
+   }
+
+Registering a test case
+-----------------------
+
+A new test case must be registered to the test_case_data section using lib_section_vars.
+The NRF_SECTION_ITEM_REGISTER macro below places the variable test_hash in a section named "test_case_data", which is initialized by main::
+
+   NRF_SECTION_ITEM_REGISTER(test_case_data, test_case_t test_hash) =
+   {
+       .p_test_case_name = "Hash test",        /**< Pointer to hash test case name. */
+       .setup = setup_test_case_hash,          /**< Pointer to hash test case setup function. */
+       .exec = exec_test_case_hash,            /**< Pointer to hash test case execute function. */
+       .teardown = teardown_test_case_hash     /**< Pointer to hash test case teardown function. */
+   };
+
+.. note::
+   The macro call to NRF_SECTION_ITEM_REGISTER must be done in a .c file.
+
+Test vectors
+------------
+
+A test vector is a set of inputs and expected outputs to verify the functionality provided in a test case.
+The ``p_test_vector_name`` variable is the only required member in the test_vector structures::
```

Comment:
Remove The line "The ``p_test_vector_name`` as it is unneeded

Reply:
done